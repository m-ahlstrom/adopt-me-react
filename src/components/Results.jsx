import { useState, useMemo } from 'react'
import Pet from './Pet'
import Pagination from './Pagination'

const Results = ({ pets }) => {
  let pageSize = 6
  let data = pets

  const [currentPage, setCurrentPage] = useState(1)

  const currentPetData = useMemo(() => {
    const firstPageIndex = (currentPage - 1) * pageSize
    const lastPageIndex = firstPageIndex + pageSize
    return data.slice(firstPageIndex, lastPageIndex)
  }, [pageSize, currentPage, data])

  return (
    <div className="m-10 grid grid-cols-1 rounded-lg bg-background-color dark:bg-zinc-700 dark:text-white p-10 shadow-lg lg:col-span-2">
      <div className="mb-4 grid grid-cols-1 gap-4 md:grid-cols-2">
        {!pets.length ? (
          <h1>No Pets Found</h1>
        ) : (
          currentPetData.map((pet) => {
            return (
              <Pet
                animal={pet.animal}
                key={pet.id}
                name={pet.name}
                breed={pet.breed}
                images={pet.images}
                location={`${pet.city}, ${pet.state}`}
                id={pet.id}
              />
            )
          })
        )}
      </div>
      <Pagination
        className="pagination-bar"
        currentPage={currentPage}
        totalCount={data.length}
        pageSize={pageSize}
        onPageChange={(page) => setCurrentPage(page)}
      />
    </div>
  )
}

export default Results
