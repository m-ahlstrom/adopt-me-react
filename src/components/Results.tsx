import { useState, useMemo } from 'react'
import Pet from './Pet'
import { Pet as IPet } from '../APIResponseTypes'
import Pagination from './Pagination'

const Results = ({ pets }: { pets: IPet[] }) => {
  const pageSize = 6
  const data = pets

  const [currentPage, setCurrentPage] = useState(1)

  const currentPetData = useMemo(() => {
    const firstPageIndex = (currentPage - 1) * pageSize
    const lastPageIndex = firstPageIndex + pageSize
    return data.slice(firstPageIndex, lastPageIndex)
  }, [pageSize, currentPage, data])

  return (
    <div className="m-10 grid grid-cols-1 rounded-lg bg-background-color p-10 shadow-lg dark:bg-zinc-700 dark:text-white lg:col-span-2">
      <div className="mb-4 grid grid-cols-1 gap-4 md:grid-cols-2">
        {!pets.length ? (
          <h1>No Pets Found</h1>
        ) : (
          currentPetData.map((pet) => {
            return (
              <Pet
                animal={pet.animal}
                key={pet.id}
                name={pet.name}
                breed={pet.breed}
                images={pet.images}
                location={`${pet.city}, ${pet.state}`}
                id={pet.id}
              />
            )
          })
        )}
      </div>
      <Pagination
        currentPage={currentPage}
        totalCount={data.length}
        pageSize={pageSize}
        onPageChange={(page: number) => setCurrentPage(page)}
      />
    </div>
  )
}

export default Results
