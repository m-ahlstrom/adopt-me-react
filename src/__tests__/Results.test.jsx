import { expect, test } from 'vitest'
import { render } from '@testing-library/react'
import Results from '../components/Results.tsx'

test('renders correctly with no pets', () => {
  const { asFragment } = render(<Results pets={[]} />)
  expect(asFragment()).toMatchSnapshot()
})
