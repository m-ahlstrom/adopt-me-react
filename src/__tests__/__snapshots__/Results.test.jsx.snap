// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders correctly with no pets 1`] = `
<DocumentFragment>
  <div
    class="m-10 grid grid-cols-1 rounded-lg bg-background-color p-10 shadow-lg dark:bg-zinc-700 dark:text-white lg:col-span-2"
  >
    <div
      class="mb-4 grid grid-cols-1 gap-4 md:grid-cols-2"
    >
      <h1>
        No Pets Found
      </h1>
    </div>
  </div>
</DocumentFragment>
`;
